pkg/build/nodeimage/internal/container/docker/pull.go

```diff
@@ -19,25 +19,29 @@ package docker
 import (
 	"time"
 
+	"k8s.io/apimachinery/pkg/util/wait"
 	"sigs.k8s.io/kind/pkg/exec"
 	"sigs.k8s.io/kind/pkg/log"
 )
 
 // Pull pulls an image, retrying up to retries times
 func Pull(logger log.Logger, image string, platform string, retries int) error {
-	logger.V(1).Infof("Pulling image: %s for platform %s ...", image, platform)
-	err := exec.Command("docker", "pull", "--platform="+platform, image).Run()
-	// retry pulling up to retries times if necessary
-	if err != nil {
-		for i := 0; i < retries; i++ {
-			time.Sleep(time.Second * time.Duration(i+1))
-			logger.V(1).Infof("Trying again to pull image: %q ... %v", image, err)
-			// TODO(bentheelder): add some backoff / sleep?
-			err = exec.Command("docker", "pull", "--platform="+platform, image).Run()
-			if err == nil {
-				break
-			}
+	backoff := wait.Backoff{
+		Duration: time.Second,
+		Factor:   2,
+		Steps:    retries,
+	}
+	var lastErr error
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
I don't think exponential backoff makes sense here, we're expecting some small network flake, users do not want to wait exponential time for an image pull, we're just *slightly* optimizing vs a plain `docker run`.